### PR TITLE
Express console code refactor, cleanup and fixes

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Rewrites a good chunk of express console code, removing one letter variables, duplicate code and straightening all the staircases with early returns. RIP cooldowns in ui_data()

Mapped-in consoles now actually work, as currently they initialize before SSshuttles which results in them having a completely empty packs menu. This is fixed by making them report themselves to SSshuttles which will call package init on them once it itself initializes.

Emagging the console will update the prices in the UI, and emagged consoles will try to send at least 1 package instead of requiring the budget to have enough funds to send all 5 pods. Because when you press a button with a price tag on it, you'd expect at least something to happen.

Beacons now reset their offsets when anchored because thats been triggering my OCD. Upgrade disks aren't deleted and instead drop when the console is deconstructed.

## Why It's Good For The Game

Code is crunchy, stuff is broken, prices aren't displayed correctly when the console is emagged and you'd expect at least something to arrive when you press a button with a price that you can afford.

## Changelog
:cl:
fix: Mapped in express supply consoles now work instead of displaying an empty UI.
fix: Emagged express supply consoles now display updated prices.
refactor: Rewrote a large chunk of express supply console code
balance: Express supply consoles now drop their upgrade disk upon being deconstructed, and emagged consoles now will try to send at least one package to the station if cargo budget doesn't have enough funds for all 5.
/:cl:

### CODE REVIEW

File: code/modules/cargo/expressconsole.dm [hunk 1]
```diff
@@ -33,24 +46,30 @@
...
-	else if(istype(W, /obj/item/supplypod_beacon))
-		var/obj/item/supplypod_beacon/sb = W
-		if (sb.express_console != src)
-			sb.link_console(src, user)
-			return TRUE
-		else
-			to_chat(user, span_alert("[src] is already linked to [sb]."))
-	..()
+		tool.forceMove(src)
+		return ITEM_INTERACT_SUCCESS
```
Comment (00-Steven) [inline code/modules/cargo/expressconsole.dm:63]: may be good to block more disks from being inserted after the first one

**Comment by carlarctg** [comment]
this isnt a balance change

File: code/modules/cargo/expressconsole.dm [hunk 2]
```diff
@@ -11,18 +14,28 @@
...
+	/// Cooldown to prevent beacon spam
+	COOLDOWN_DECLARE(beacon_print_cooldown)
 
 /obj/machinery/computer/cargo/express/Initialize(mapload)
 	. = ..()
 	packin_up()
+	landingzone = GLOB.areas_by_type[landingzone]
+	if (isnull(landingzone))
+		WARNING("[src] couldnt find a Quartermaster/Storage (aka cargobay) area on the station, and as such it has set the supplypod landingzone to the area it resides in.")
+		landingzone = get_area(src)
```
Comment (MrMelbert) [inline code/modules/cargo/expressconsole.dm:38]: Should we be checking for `mapload` here? 
If not, should this be a stack trace instead?

File: code/modules/cargo/expressconsole.dm [hunk 3]
```diff
@@ -33,24 +46,33 @@
 		beacon.unlink_console()
 	return ..()
 
-/obj/machinery/computer/cargo/express/attackby(obj/item/W, mob/living/user, params)
-	if(W.GetID() && allowed(user))
+/obj/machinery/computer/cargo/express/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = ..()
```
Comment (MrMelbert) [inline code/modules/cargo/expressconsole.dm:50]: Parent call does nothing I believe

**Comment by Ghommie** [comment]
immagine

**Pull request merged**